### ISSUE TITLE
Observational weights for column_data.py

### ISSUE THREAD

**Issue by pford221** [fastai/fastai#606]
Hello,

If I worked and submitted a PR for incorporating observational weights (i.e. each observation can have a potentially different influence on the loss function) would it have a chance of being approved? Observational weights can be important for structured data whereby a certain record has been observed for a long period time than another record and therefore carries more information. This is very common when working with insurance data for instance where any one record can represent a varying length of an insurance contract.

I think I know how to incorporate observational weights without too much detritus but it will require some extra conditionals here and there in `fit()` and `Stepper`. It will also require users to write custom pytorch loss functions and evaluation metric functions since pytorch doesn't support observational weights.

Let me know and I'd be happy to make a pass at this.

Thanks,
Patrick

**Comment by sgugger** [comment]
Hi there, thanks for wanting to contribute to the library!

Every new development is welcome as long as it helps getting better result and doesn't interfere with pre-existing code. Ideally if you can keep the changes in the fit function to their minimum and use a separate custom Stepper class that is a child of the current Stepper class, that would be best.

A notebook that shows the benefits of this, with an example of use with the custom loss/evaluation metrics would be perfect and maximize your chances at getting your PR quickly approved.
